router/io-spec.gobra

```diff
@@ -469,25 +480,15 @@ pure func midSeg(
 
 ghost
 requires dp.Valid()
-requires len(raw) > 4
-requires acc(sl.AbsSlice_Bytes(raw, 0, len(raw)), _)
-requires unfolding acc(sl.AbsSlice_Bytes(raw, 0, len(raw)), _) in
-	let hdr := binary.BigEndian.Uint32(raw[0:4]) in
-	let metaHdr := scion.DecodedFrom(hdr) in
-	let seg1 := int(metaHdr.SegLen[0]) in
-	let seg2 := int(metaHdr.SegLen[1]) in
-	let seg3 := int(metaHdr.SegLen[2]) in
-	let base := scion.Base{metaHdr,
-		numInfoFields(seg1, seg2, seg3),
-		seg1 + seg2 + seg3} in
-	metaHdr.InBounds() &&
-	0 < metaHdr.SegLen[0] &&
-	base.ValidCurrInfSpec() &&
-	base.ValidCurrHfSpec() &&
-	len(raw) >= pktLen(seg1, seg2, seg3)
+requires acc(sl.AbsSlice_Bytes(raw, 0, len(raw)), R55)
+requires validPktMetaHdr(raw)
+//ensures res != none[io.IO_pkt2] ==> len(get(res).CurrSeg.Future) > 0
+//ensures res != none[io.IO_pkt2] ==> get(res).CurrSeg.Future[0].extr_asid() == dp.Asid()
```

Comment:
Why are these posts here? Is the goal to prove them before we merge? Maybe sth that might make sense here is to prove these conditions extrinsically (i.e., on a separate lemma) and call the lemma when we need them.

Reply:
Yes if we want to keep everything opaque, we have to prove them separately. I removed them for now.